### PR TITLE
EDSC-3240: Adds UMM-T smart handoffs capability

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: static/src/js/actions/focusedCollection.js [hunk 1]
```diff
@@ -151,6 +151,14 @@ export const getFocusedCollection = () => async (dispatch, getState) => {
             query
           }
         }
+        tools {
+          count
+          items {
+            conceptId
```
Comment (macrouch) [inline static/src/js/actions/focusedCollection.js:157]: Is this necessary?

File: static/src/js/util/handoffs/getHandoffValue.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    spatial = {}
+  } = collectionQuery
+
+  const { endDate, startDate } = temporal
+
+  let value = ''
+
+  // Bounding Box value
+  if (valueType === 'https://schema.org/box') {
+    const { boundingBox } = spatial
```
Comment (macrouch) [inline static/src/js/util/handoffs/getHandoffValue.js:33]: Use MBR

File: static/src/js/util/handoffs/getHandoffValue.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    const { sotoLayers = [] } = handoffs
+    const { tags } = collectionMetadata
+    const gibsOptions = getValueForTag('gibs', tags)
+
+    if (gibsOptions) {
+      const gibsLayers = gibsOptions.map(data => data.product)
+
+      // Filter out layers that are not included in SOTO's capabilities
+      const includedSotoLayers = gibsLayers.filter(layer => sotoLayers.includes(layer))
+      value = includedSotoLayers.map(data => `${data}(la=true)`)
```
Comment (macrouch) [inline static/src/js/util/handoffs/getHandoffValue.js:62]: Comment this to explain the SOTO specific stuff

File: static/src/js/util/handoffs/generateHandoffs.js [hunk 4]
```diff
@@ -1,28 +1,117 @@
...
+      }
+
+      // Add the value to be expanded onto the urlTemplate later
+      urlValues[valueName] = value
+    })
+
+    // If all the required inputs are present, push the generated link onto handoffLinks to be returned
+    if (allRequiredItemsPresent) {
+      handoffLinks.push({
+        title: name,
```
Comment (macrouch) [inline static/src/js/util/handoffs/generateHandoffs.js:83]: Use UMM-T LongName

File: static/src/js/actions/focusedCollection.js [hunk 5]
```diff
@@ -244,6 +254,12 @@ export const getFocusedCollection = () => async (dispatch, getState) => {
 
         // Query CMR for granules belonging to the focused collection
         dispatch(actions.getSearchGranules())
+
+        // If any tools are SOTO, call fetchSotoLayers
+        const { items } = tools
+        if (items && items.filter(item => item.name.toLowerCase() === 'soto').length > 0) {
```
Comment (abbottry) [inline static/src/js/actions/focusedCollection.js:260]: This seems like a slightly more difficult to read version of `.some()`

File: static/src/js/actions/handoffs.js [hunk 6]
```diff
@@ -0,0 +1,39 @@
+import { SET_SOTO_LAYERS } from '../constants/actionTypes'
```
Comment (abbottry) [inline static/src/js/actions/handoffs.js:1]: We've historically sorted these a bit more by type of import.

File: static/src/js/containers/ProjectCollectionsContainer/ProjectCollectionsContainer.js [hunk 7]
```diff
@@ -9,6 +9,7 @@ import { metricsDataAccess } from '../../middleware/metrics/actions'
 import { getProjectCollectionsMetadata, getProjectCollectionsIds } from '../../selectors/project'
 
 import ProjectCollections from '../../components/ProjectCollections/ProjectCollections'
+import { getHandoffs } from '../../selectors/handoffs'
```
Comment (abbottry) [inline static/src/js/containers/ProjectCollectionsContainer/ProjectCollectionsContainer.js:12]: Move up above to similar import types

File: static/src/js/containers/SearchPanelsContainer/SearchPanelsContainer.js [hunk 8]
```diff
@@ -10,6 +10,7 @@ import { getFocusedGranuleMetadata } from '../../selectors/granuleMetadata'
 import { metricsCollectionSortChange } from '../../middleware/metrics/actions'
 import actions from '../../actions/index'
 import SearchPanels from '../../components/SearchPanels/SearchPanels'
+import { getHandoffs } from '../../selectors/handoffs'
```
Comment (abbottry) [inline static/src/js/containers/SearchPanelsContainer/SearchPanelsContainer.js:13]: Move up to similar types imports

**Review by abbottry — COMMENTED** [review]
Looks pretty good, we do template parsing in the CWIC code as well, we should write a ticket to swap that code over to using this library if you find it useful.

**Review by trevorlang — APPROVED** [review]

**Review by abbottry — APPROVED** [review]

**Pull request merged**